src/algo/ClosedHash.js

```diff
@@ -704,22 +705,23 @@ export default class ClosedHash extends Hash {
 	}
 
 	clear() {
-		this.commands = [];		
-		
+		this.commands = [];
+
 		for (let i = 0; i < this.table_size; i++) {
 			this.cmd(act.delete, this.hashTableVisual[i]);
 			this.cmd(act.delete, this.indexLabelID[i]);
 		}
 
-		this.table_size = CLOSED_HASH_TABLE_SIZE;
+		this.table_size = parseInt(this.initialSizeField.value)
+			? Math.min(Math.max(0, parseInt(this.initialSizeField.value)), MAX_SIZE)
+			: CLOSED_HASH_TABLE_SIZE;
```

Comment:
It would be good to validate the load factor here. If the new table_size is greater than max size, set the load factor to .99 so that another resize won't happen. 

It actually can't resize beyond 70 because of a check inside insertElement(), but we want to display that functionality to the user through the load factor.

Reply:
Something else to add (not related to your changes) is updating the load factor with every call to insertElement(). This is currently done in OpenHash (line 208) and dismisses the "Max Array Length) warning, I just think it looks nicer :D

Reply:
Yeah, I made the table_size validations 👍  I also disabled the button (and reenabled it when a smaller capacity was given as input) to let the user know they can't change it, although it's less likely to be used.